user_sync/config.py

```diff
@@ -354,8 +354,13 @@ def get_rule_options(self):
                     self.logger.warning("Unrecognized extension context '%s' ignored", context)
         
         options = self.options
+        if (options['directory_group_mapped'] == True):
+            directory_group_filter = set(self.get_directory_groups().iterkeys())
```

Comment:
This is fine...  tho if calling get_directory_groups is slow, you can do everything in app.py beginWork instead, after get_directory_groups and get rule_options are called and you override the rule_options there.